### PR TITLE
Fix FBXBaker not properly handling textures with the same name

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by Nex-Pro** [comment]
@huffman Wow you got onto that fast

File: libraries/baking/src/FBXBaker.cpp [hunk 1]
```diff
@@ -583,48 +583,56 @@ void FBXBaker::rewriteAndBakeSceneTextures() {
...
 
                                 if (!_bakingTextures.contains(urlToTexture)) {
+                                    if (textureFileInfo.suffix() == BAKED_TEXTURE_EXT.mid(1)) {
+                                        // re-baking an FBX that already references baked textures is a fail
+                                        // so we add an error and return from here
+                                        handleError("Cannot re-bake a file that references compressed textures");
+
+                                        return;
+                                    }
+
```
Comment (birarda) [inline libraries/baking/src/FBXBaker.cpp]: Can't we still check this at the top?

File: libraries/baking/src/TextureBaker.cpp [hunk 2]
```diff
@@ -26,15 +26,19 @@
...
     _originalTexture(textureContent),
     _textureType(textureType),
-    _outputDirectory(outputDirectory)
+    _outputDirectory(outputDirectory),
+    _bakedTextureFileName(bakedFilename)
 {
-    // figure out the baked texture filename
-    auto originalFilename = textureURL.fileName();
-    _bakedTextureFileName = originalFilename.left(originalFilename.lastIndexOf('.')) + BAKED_TEXTURE_EXT;
+    if (bakedFilename.isEmpty() || bakedFilename.isNull()) {
```
Comment (birarda) [inline libraries/baking/src/TextureBaker.cpp]: A null string is also empty, can just do the one check here

**Comment by huffman** [comment]
@Nex-Pro Motivated to get it into this next release :)

**Comment by huffman** [comment]
Test plan:

1. Baking assets should still work as expected
2. @Nex-Pro's bale model should now bake successfully

**Comment by Nex-Pro** [comment]
Testing this by request when built.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3523/HighFidelity-Beta-PR11429-26f6a3c3f0cf1c186400eeba23cdf07254183261.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3523/HighFidelity-Beta-PR11429-26f6a3c3f0cf1c186400eeba23cdf07254183261.exe

**Comment by Nex-Pro** [comment]
T e s t i n g  this

**Comment by Nex-Pro** [comment]
@huffman @birarda win build link is bad

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3528/HighFidelity-Beta-PR11429-26f6a3c3f0cf1c186400eeba23cdf07254183261.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3528/HighFidelity-Beta-PR11429-26f6a3c3f0cf1c186400eeba23cdf07254183261.exe

**Comment by Nex-Pro** [comment]
Ok , here we go !

**Comment by Nex-Pro** [comment]
## PASS

Tried this with a few models and it works fine.  Baked a more complex model and two cores of my i7 were almost saturated while the baking took place. 

The bale model uses the same texture in two slots and worked.
It also has two non-textured materials in it and they worked also.

Saw an issue unrelated to this PR and will investigate.

**Pull request merged**